Bugfix: `Worksheet.addSheetData` corrupting Xlsx file saved in Excel before

omaus opened the pull request:
- Fixes #16 
- Indirectly fixes https://github.com/nfdi4plants/ISADotNet/issues/49
- Adds `setSheetData` function
- Adds `replaceSheetData` function
- Some typos fixed, phrasing and wording unified

omaus committed:
Fix `addSheetData` bug :beetle:

omaus committed:
Add `setSheetData` function

omaus committed:
Fix typos, unify wording :lipstick:

omaus committed:
Add `replaceSheetData` function, fix typos, unify wording/phrasing :sparkles: :lipstick:

HLWeil commented on src/FSharpSpreadsheetML/Workbook.fs:
```diff
@@ -123,3 +123,21 @@ module WorkbookPart =
         workbookPart
+
+    /// Replaces the SheetData of the Sheet with the given sheetname.
+    let replaceSheetData (sheetName : string) (data : SheetData) (workbookPart : WorkbookPart) =
```
Rename to "replaceSheetDataByName"

omaus committed:
Rename `replaceSheetData` to `replaceSheetDataByName` :art:

As requested in https://github.com/CSBiology/FSharpSpreadsheetML/pull/17#pullrequestreview-811381547
